torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,221 @@
+        qCofnigDict: dictionary that maps from name of submodule to quantization
+                     configuration
+    """
+    addObserver(modInstance, qConfigDict)
+    addQuantDeQuant(modInstance, qConfigDict)
+
+class AbstractQuant(nn.Module):
+    r"""Abstract quantize module which will be replace to actual Quantize module
+    in swapModule function
+    """
+    def __init__(self, qconfig):
+        super(AbstractQuant, self).__init__()
+        self.add_module('observer', qconfig.activation())
+
+    def forward(self, x):
+        self.observer(x)
+        return x
+
+class AbstractDeQuant(nn.Module):
+    r"""Abstract dequantize module which will be replace to actual DeQuantize module
+    in swapModule function
+    """
+    def __init__(self):
+        super(AbstractDeQuant, self).__init__()
+
+    def forward(self, x):
+        return x
+
+def calculateQParams(MyModule):
+    r""" Calculates Quantizer parameters for activation tensors based on observer statistics
+    Updates the qparams parameter for each module with the quantizer params.
+
+    Args:
+        MyModule: Model with observer stats
+    """
+    if hasattr(MyModule, 'observer'):
+        if isinstance(MyModule.observer, Observer):
+            # Simple symmetric quantization over entire observed range
+            MyModule.qparams = MyModule.observer.calculate_qparams()
+    for Module in MyModule.children():
+        calculateQParams(Module)
+
+def quantize(module, qConfigDict, eval_fn, *eval_args):
+    r"""Given a float module and qConfigDict, convert it to a quantized module
+
+    First it will prepare the module to add observer and abstract quant
+    deqaunt modules, then it calls `eval_fn` which will run the calibration
+    step and observers will record the stats of the tensors, after that we will
+    call `convert` which will convert the module to a quantized module.
+    """
+    prepare(module, qConfigDict)
+    eval_fn(module, *eval_args)
+    convert(module)
+    return
+
+def convert(module):
+    r""" Utility function to traverse module tree and swap modules marked with
+         AbstractQuant and AbstractDeQuant with
+         quantized versions.
+    """
+    calculateQParams(module)
+    return convertToQuantized(module)
+
+def convertToQuantized(module):
+    r"""Convert a module with qparams to a quantized version of the module
+    """
+    module_swapped = swapModule(module)
+    if len(list(module.named_children())) == 0:
+        return module_swapped
+
+    reassign = {}
+    for name, mod in module.named_children():
+        new_mod = convertToQuantized(mod)
+        if new_mod is not mod:
+            reassign[name] = new_mod
+
+    for name, mod in reassign.items():
+        setattr(module_swapped, name, mod)
+
+    return module_swapped
+
+DEFAULT_MODULE_MAPPING = {
+    torch.nn.Linear: nnq.Linear,
+    torch.nn.ReLU: nnq.ReLU,
+}
+
+ABSTRACT_MODULE_MAPPING = {
+    AbstractQuant: nnq.Quantize,
```

Comment:
QuantStub or QuantSpot or something like that might be a better name

Reply:
changed to Stub